### PR TITLE
VTOL back transition acceptance radius calculation

### PR DESCRIPTION
**Pull request by sanderux**
Calculate acceptance radius based on groundspeed and duration for back transition in VTOL_LAND to prevent overshoot of landing waypoint.

ref https://github.com/PX4/Firmware/issues/5626 https://github.com/PX4/Firmware/issues/5681

/cc @AndreasAntener @tumbili

### CODE REVIEW

File: src/modules/navigator/mission_block.cpp [hunk 1]
```diff
@@ -307,6 +308,19 @@ MissionBlock::is_mission_item_reached()
 				mission_acceptance_radius = _navigator->get_acceptance_radius();
 			}
 
+			/* for vtol back transition calculate acceptance radius based on time and ground speed */
+			if (_mission_item.vtol_back_transition) {
+
+			    float groundspeed = sqrtf(_navigator->get_global_position()->vel_n * _navigator->get_global_position()->vel_n +
```
Comment (AndreasAntener) [inline src/modules/navigator/mission_block.cpp]: The problem is that the back transition time is not right for VTOLs flying with airspeed. We've just started testing another option that you could just add here instead, using a scaling factor for the already ground speed dependent dynamic radius:
```
			/* scale acceptance radius if requested */
			if (_mission_item.vtol_acceptance_radius) {
				mission_acceptance_radius *= _param_vtol_bt_rad_scale.get();
			}
```

Comment (AndreasAntener) [inline src/modules/navigator/mission_block.cpp]: `vtol_acceptance_radius` would be your `vtol_back_transition`

Comment (sanderux) [inline src/modules/navigator/mission_block.cpp]: How do you enable wind in sitl?

Comment (dagar) [inline src/modules/navigator/mission_block.cpp]: @sanderux set this option https://github.com/PX4/sitl_gazebo/blob/master/CMakeLists.txt#L146

Can be modified here - https://github.com/PX4/sitl_gazebo/blob/master/models/standard_vtol/standard_vtol.sdf#L853

**Comment by sanderux** [comment]
Rebased, i will try to get a flight window soon

**Comment by dagar** [comment]
Architecturally I don't like that navigator needs to have intimate knowledge of a specific vehicle. Also, mission item was optimized and squashed just to save space and now we have an extra field that only applies for a tiny portion of a specific vehicle's mission. 

If this solves the problem for now I'll stop complaining, but it really makes me wonder if we could do a lot better with #4845.

**Comment by sanderux** [comment]
Works awesome
image

**Comment by sanderux** [comment]
Rebased and re-sitl tested.
@AndreasAntener can you review?

**Comment by dagar** [comment]
Could we get away with setting _mission_item.acceptance_radius = gndspeed * vt_b_trans_dur where the mission item is created?

**Comment by sanderux** [comment]
Wouldnt work. The groundspeed  can change significantly at the point between reading the item and reaching the waypoint.

**Comment by sanderux** [comment]
@LorenzMeier should be ready for merge after pass

**Comment by sanderux** [comment]
@LorenzMeier i re-checked the last mission, it was wind that blew it off course, the math works.
Could you merge this?

**Pull request merged**